src/uu/sort/src/sort.rs

```diff
@@ -731,20 +738,50 @@ fn get_nums_dedup(a: &str) -> &str {
         ""
     // Prepare lines for comparison of only the numerical leading numbers
     } else {
-        get_leading_num(s)
+        let result = match settings.mode {
+            SortMode::Numeric => get_leading_num(s),
+            SortMode::GeneralNumeric => get_leading_gen(s),
+            SortMode::HumanNumeric => get_leading_num(s),
+            SortMode::Version => get_leading_num(s),
```

Comment:
You could use the `|` syntax for patterns here (https://doc.rust-lang.org/nightly/book/ch18-03-pattern-syntax.html#multiple-patterns) and do `SortMode::Numeric | SortMode::HumanNumeric | SortMode::Version => get_leading_num(s)`.

Reply:
Thanks!